docs/yul.rst

```diff
@@ -559,8 +559,7 @@ regular strings in native encoding. For code,
 
 Grammar::
 
-    TopLevelObject = 'object' '{' Code? ( Object | Data )* '}'
-    Object = 'object' StringLiteral '{' Code? ( Object | Data )* '}'
+    Object = 'object' StringLiteral '{' Code ( Object | Data )* '}'
```

Comment:
Why is code not optional?

Reply:
I think it should be `'object' StringLiteral '{' (Code | Object | Data )+ '}'

Reply:
I would not allow multiple code blocks - what happens to control flow at the end of the first code block? If you want to refer to code as data, you should wrap it inside a sub-object.

About objects without code: Do you see a use-case? We can allow it at some later point, if we want, but for now, I would require code - the code itself can be empty, though.